### PR TITLE
Move to src-based layout, declare typed (PEP 561), modernize type annotations (and more)

### LINKED ISSUE #110: mypy error: Skipping analyzing "crc": [...] missing library stubs or py.typed marker
Thanks for having type annotation in your package! However, I can't use it with type checkers in my own code, because `crc` is missing a `py.typed` marker file. Please see [PEP 561] for more information.

This leads to errors in downstream projects using type checkers like mypy:

```
error: Skipping analyzing "crc": module is installed, but missing library stubs or py.typed marker  [import]
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

Looking at your source code package...:

```console
$ tar -tf crc-4.3.0.tar.gz 
crc-4.3.0/LICENSE.txt
crc-4.3.0/README.md
crc-4.3.0/crc.py
crc-4.3.0/pyproject.toml
crc-4.3.0/setup.py
crc-4.3.0/PKG-INFO
```

... I understand your current distribution is single-file/module-only, but it seems that PEP requires a refactor into a package structure.

> This PEP does not support distributing typing information as part of module-only distributions or single-file modules within namespace packages.
>
> The single-file module should be refactored into a package and indicate that the package supports typing as described above.

Let me know if you need some help with that; I'm happy to provide a PR.

[PEP 561]: https://peps.python.org/pep-0561/

### PR DESCRIPTION
**Pull request by gertvdijk**
Fixes #110.

### CODE REVIEW

File: src/crc/_crc.py [hunk 1]
```diff
@@ -10,36 +11,44 @@
...
-    ByteString,
+    Final,
     Iterable,
     Iterator,
-    List,
-    Optional,
     Union,
 )
 
 __author__ = "Nicola Coretti"
```
Comment (Nicoretti) [inline src/crc/_crc.py:21]:
```suggestion
__authors__ = ["Nicola Coretti", "Gert van Dijk"]
```

Comment (Nicoretti) [reply src/crc/_crc.py:21]: I also plan to add a contributors page to the docs and add you if that's fine with you.

Comment (gertvdijk) [reply src/crc/_crc.py:21]: I think the `__author__` remains singular even if it's a list of authors. Well, in fact there seems to be no standard on this, but this flake8 plugin says so. 😄  https://pypi.org/project/flake8-author/

A contributors page is cool, appreciated.

**Comment by gertvdijk** [comment]
(update: adjusted some more settings for mkdocs and adopted that change to `__author__`.)

The PR is large and scope is getting broad, so, if you disagree to some of the changes, feel free to cherry-pick some commits and skip others for later. 😃

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**